bot/utils/helpers.py

```diff
@@ -0,0 +1,55 @@
+import calendar
+from datetime import datetime
+from typing import Literal, Union
+
+import discord
+from discord import NotFound
+
+from bot.consts import Colors
+from bot.messaging.events import Events
+from bot.sock_bot import SockBot
+
+FORMAT = '%Y-%m-%d %H:%M:%S'
+
+
+def strtodt(date: str) -> datetime:
+    """A simple util for translating UTC timestamps in the database to UTC datetime.datetime objects."""
+    return datetime.strptime(date, FORMAT)
+
+
+def as_timestamp(date: datetime, /, style: Literal["f", "F", "d", "D", "t", "T", "R"] = "f") -> str:
+    """
+    Formats the given datetime to a Discord timestamp.
+    Used over discord.utils.format_dt due to incorrect timestamp output.
+    """
+    timestamp = calendar.timegm(date.utctimetuple())
+    return f'<t:{timestamp}:{style}>'
```

Comment:
This is already part of discord.py https://discordpy.readthedocs.io/en/stable/api.html#discord.utils.format_dt

Reply:
Their method actually ends up using GMT instead of UTC as the timezone, which heavily skews the actual timestamp Discord will display, so this is why this method exists.

Reply:
<img width="858" alt="image" src="https://user-images.githubusercontent.com/38477514/221326920-82f55324-6db2-4c3f-9d8b-f9dee488b2ba.png">
![](https://cdn.discordapp.com/emojis/1043952866282442863.gif?size=240&quality=lossless)

Reply:
fake news. i'll go ahead and test..........aha ![image](https://i.imgur.com/Npnr0HL.png)
```python
embed.add_field(name='Start Date (og)', value=as_timestamp(current_semester.start_date))
embed.add_field(name='End Date (og)', value=as_timestamp(current_semester.end_date))
embed.add_field(name='Start Date (discord)', value=format_dt(current_semester.start_date))
embed.add_field(name='End Date (discord)', value=format_dt(current_semester.end_date))
```